autotest/pymod/uffd.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+###############################################################################
+# $Id$
+#
+# Project:  GDAL/OGR Test Suite
+# Purpose:  Python Library supporting GDAL/OGR Test Suite
+# Author:   James McClain <james.mcclain@gmail.com>
+#
+###############################################################################
+# Copyright (c) 2018, Dr. James McClain <james.mcclain@gmail.com>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+
+from osgeo import gdal
+
+
+def uffd_compare(file_name):
+    dataset1 = gdal.Open('./data/{}'.format(file_name))
+    dataset2 = gdal.Open('/vsicurl/https://github.com/OSGeo/gdal/raw/master/autotest/gdrivers/data/{}'.format(file_name))  # noqa
+    if dataset2 is None:
+        return None
+    if (dataset1.GetMetadata() != dataset2.GetMetadata()):
+        return False
+    if (dataset1.RasterCount != dataset2.RasterCount):
+        return False
+    for i in range(dataset1.RasterCount):
+        raster1 = dataset1.GetRasterBand(i+1).ReadAsArray()
```

Comment:
you can just compare GetRasterBand(i+1).Checksum()

Reply:
Done.